master/pkg/searcher/searcher.go

```diff
@@ -66,7 +66,7 @@ func (s *Searcher) TrialExitedEarly(
 		return nil, errors.Errorf("unexpected trial ID sent to searcher: %d", trialID)
 	}
 
-	s.eventLog.TrialExitedEarly(requestID)
+	s.eventLog.TrialExitedEarly(requestID, *exitedReason)
```

Comment:
blocking: I know this only gets called when the trial actor sees the `exitedReason` is non nil, so this being nil is a would only error be a programming error, but can we check it and `panic` with a really specific message if that happens?